holmes/plugins/toolsets/atlas_mongodb/mongodb_atlas.py

```diff
@@ -0,0 +1,304 @@
+import requests  # type: ignore
+import logging
+from typing import Any, Dict, Tuple, List
+from holmes.core.tools import (
+    CallablePrerequisite,
+    Tool,
+    ToolParameter,
+    Toolset,
+    ToolsetTag,
+)
+
+from pydantic import BaseModel, PrivateAttr
+from holmes.core.tools import StructuredToolResult, ToolResultStatus
+from requests.auth import HTTPDigestAuth  # type: ignore
+import gzip
+import io
+from datetime import datetime, timedelta, timezone
+import os
+from collections import Counter
+
+
+class MongoDBConfig(BaseModel):
+    public_key: str
+    private_key: str
+    project_id: str
+
+
+# https://www.mongodb.com/docs/atlas/reference/api-resources-spec/v2/
+class MongoDBAtlasToolset(Toolset):
+    name: str = "MongoDBAtlas"
+    description: str = "The MongoDB Atlas API allows access to Mongodb projects and processes. You can find logs, alerts, events, slow quereies and various metrics to understand the state of Mongodb projects."
+    docs_url: str = (
+        "https://www.mongodb.com/docs/atlas/reference/api-resources-spec/v2/"
+    )
+    icon_url: str = "https://webimages.mongodb.com/_com_assets/cms/kuyjf3vea2hg34taa-horizontal_default_slate_blue.svg?auto=format%252Ccompress"
+    tags: List[ToolsetTag] = [ToolsetTag.CORE]
+    _session: requests.Session = PrivateAttr(default=requests.Session())
+
+    def __init__(self):
+        super().__init__(
+            prerequisites=[CallablePrerequisite(callable=self.prerequisites_callable)],
+            tools=[
+                ReturnProjectAlerts(toolset=self),
+                ReturnProjectProcesses(toolset=self),
+                ReturnProjectSlowQueries(toolset=self),
+                ReturnEventsFromProject(toolset=self),
+                ReturnLogsForProcessInProject(toolset=self),
+                ReturnEventTypeFromProject(toolset=self),
+            ],
+        )
+        instructions_filepath = os.path.abspath(
+            os.path.join(os.path.dirname(__file__), "instructions.jinja2")
+        )
+        self._load_llm_instructions(jinja_template=f"file://{instructions_filepath}")
+
+    def prerequisites_callable(self, config: dict[str, Any]) -> Tuple[bool, str]:
+        if not config:
+            return False, "Missing config credentials."
+
+        try:
+            self.config: Dict = MongoDBConfig(**config).model_dump()
+            self._session.headers.update(
+                {"Accept": "application/vnd.atlas.2025-03-12+json"}
+            )
+            self._session.auth = HTTPDigestAuth(
+                self.config.get("public_key"),
+                self.config.get("private_key"),
+            )
+            return True, ""
+        except Exception:
+            logging.exception("Failed to set up MongoDBAtlas toolset")
```

Comment:
Please add here the reason, e.g. if the creads are invalid than invalid authnticaiton. If there was some issue with config parsing add it as well. We use the error message to show it in the UI.